articles/file-handling-in-Cpp/index.md

```diff
@@ -0,0 +1,176 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/file-handling-in-Cpp/
+title: File Handling in C++
+description: In this article, we will walk through how to implement the File Handling function in C++, and we will cover the key working pointers of file handling.
+author: parampreet-singh
+date: 2020-09-21T00:00:00-12:00
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/file-handling-in-Cpp/hero.jpg
+    alt: functions Function C++ cpp
+---
+If you are a Software Engineer, you may come across File handling or file input/output at some point in your career. It is the ability to store the output and perform various operations on it, such as updating the data or reading/writing text to a file. Files are always a vital way to store data.
+<!--more-->
+### Why do you need File Handling?
+- **Reusability**: Generally, developers work with big data, and it's always not efficient to write everything out by storing data in files, one can use those files anytime.
+
+- **Larger Storage Capacity**: Storing a large amount of data or test cases (to make sure that your code runs in every condition) is a tedious task, and by using files, a developer can store that data easily.
+
+- **Portability**: With files, developers can transfer the data without worrying about data loss.
+
+In this article, we will walk through how to implement a file handling function in C++, and we will cover the key properties and challenges when it comes to file handling.
+
+- Opening a File
+- Writing to a File
+- Reading from a File
+- Closing and Saving a file
+
+To perform file handling functions, we need to add another library: fstream. This library holds all the functions related to handling files. Here's how to include it.
+
+`#include<fstream>`
+
+Let's dive straight into learning.
+
+### Opening a File
+One of the most common actions in File Handling is opening a file. Let's see how to do it:
+
+```C
+ifstream fin;
+fin.open("index.txt");
+```
+
+That was simple as we just used the function **open** to open a file name "index.txt." Here we use the ` ifstream` object to open our file.
```

Comment:
That was simple as we just used the function -> That was simple. We just used the function